core/templates/dev/head/domain/exploration/StatesObjectFactory.js

```diff
@@ -62,6 +63,7 @@ oppia.factory('StatesObjectFactory', [
     };
     States.prototype.renameState = function(oldStateName, newStateName) {
       this._states[newStateName] = angular.copy(this._states[oldStateName]);
+      this._states[newStateName].name = newStateName;
```

Comment:
No, don't attach properties randomly to dicts. That's not the right way to modify them. Use a set() method instead (though I don't think the State should know its name in the first place, right?)

Reply:
Ok, will use a setName().
Also, about the name,  it seems it is being used in a lot of places. Shall I modify that later on?

Reply:
Done. (used setName()).

Reply:
Yep, could you file an issue for removing the name? OK to do that in a separate PR, but we should track it.

Reply:
Done, created [issue](https://github.com/oppia/oppia/issues/5389).